src/main/java/com/google/api/codegen/config/FieldConfig.java

```diff
@@ -212,10 +212,14 @@ public FieldConfig withResourceNameConfig(ResourceNameConfig resourceNameConfig)
   public boolean requiresParamTransformation() {
     return getResourceNameConfig() != null
         && getMessageResourceNameConfig() != null
-        && getMessageResourceNameConfig().getResourceNameType() != ResourceNameType.ANY
         && !getResourceNameConfig().equals(getMessageResourceNameConfig());
   }
 
+  public boolean requiresAnyParamTransformation() {
```

Comment:
The function name here implies something different from the implementation - maybe `requiresParamTransformationFromAny` would be clearer. This is indeed difficult because the "any" term can easily be interpreted differently from the proto meaning. 

Thinking about this further: the naming pattern "requires param transformation" is somewhat opaque, given the intent. I think I would favor a more direct approach with an enum (combine this method with `requiresParamTransformation`):

```
public TestAssertionTransformation getTestAssertionTransformation() {
  ...
}

public enum TestAssertionTransformation {
  CreateOneOfObject,
  ConvertToString
}
```

@michaelbausor what do you think?